packages/ag-charts-community/src/chart/interaction/regionManager.ts

```diff
@@ -261,12 +251,16 @@ export class RegionManager {
                 this.blockNextClickEvent = false;
                 break;
             }
-            case 'hover': {
-                if (!this.isDragMoving) {
+            case 'leave':
+            case 'enter':
+                if (!this.isDragMoving || this.deferredDragStart != null) {
```

Comment:
I'm not 100% following the logic here, but just to confirm that we won't issue drag-related events until a mouse move happens after a mouse down still, to avoid firing conflicting click and drag events on mouse-up?

Reply:
Correct. That's what the `blockNextClickEvent` state handles: We fire either 1 `'click'` event or a set of `'drag*'` events but not both.

The `case 'leave','enter':` logic stops us broadcasting which region changes when dragging. Example steps:

1. `'mousedown'` fired on the series-area.
2. `'mousemove'` fired such that the pointer leaves the bounds the series-area HTML element.
3. `'mouseleave'` fired as pointer hovers over a new HTML element (we don't want to broadcast this, because the user is still dragging).
4. `'mousemove'` fired such that the pointer returns to the bounds the series-area HTML element.
3. `'mouseenter'` fired as pointer hovers back to the series-area element (we don't want to broadcast this either; the user never really left because they are still dragging)